### ISSUE TITLE
Ability to change the buttons in the usercard

### ISSUE THREAD

**Issue by NotEchoDE** [Chatterino/chatterino2#2331]
**What should be added?**

An Ability to change the commands from the buttons in the usercard.
https://i.imgur.com/92LpF0y.png

It should be possible to edit these or to add new ones above or below them.

**Why should it be added?**
I've noticed that chatterino dont have that much moderation features yet. I am a moderator of a quite popular twitch channel and I want a feature to timout someone for "Spam" 2 Minutes or for "Some other Reason here" 5 Minutes. It would be very useful because then I don't have to add a bunch of buttons here: https://i.imgur.com/Az0hejP.png

**Comment by sando** [comment]
This functionality has already been implemented - it's available beta and nightly builds. See db3402a18ed9f2385386b9588d008e81780310d4, #2331

![image](https://user-images.githubusercontent.com/58441/103478958-45aec880-4e1e-11eb-94f9-31925da92458.png)
![image](https://user-images.githubusercontent.com/58441/103478967-4f383080-4e1e-11eb-8ce7-ca7ae9153f84.png)

**Comment by NotEchoDE** [comment]
Thats nice but it would be nice if I can set an own command aswell. Like a timeout with a reason.

**Comment by zneix** [comment]
An extension to timeout buttons specifying reasons might be cool. However, for now this can be achieved with custom moderation buttons (as seen on the screenshot above) - you can set those and those are accessible over either regular chat or usercard.

**Comment by TranRed** [comment]
I had an similar idea when I read the issue. I like how it looks. Just two things that come in to my mind:

1. This seems to be very channel specific. Like one channel you moderate might just time out spam for 1m, another one for 5m and third one for a day. I wonder whether the realisation will branch into the next request and that might lead to overcomplicating things, that should be kept relatively easy.

2. Can you differenciate between 1m "classic spam" and 1m "spoilers in chat" on the user card itself? There should be an indicator of some sort I think. Maybe be a hover/mouse over text or something if a custom reason is set? I'm unsure what would be the best for usability and/or design-wise.

**Comment by NotEchoDE** [comment]
So I just thought about it a little bit and two options that came to my mind:

1.  The timout buttons should stay where there are and there should not be any timeout reason stucked to them. They should be only contain the length of the timout. The reason should then be specified in an extra List. Here is an Example from the chat client "chatty": https://i.imgur.com/6kGIUzK.png There should alse be some hotkeys, as you see in the screenshot, to open the dropdown menu and to select a reason. Hotkeys for the timeout buttons should also be added. All hotkeys should be configuratable.

2.  The concept from @zneix is very good but I have too add some thing there. You maybe noticed that I like hotkeys so my suggestion here is to add configuratable hotkeys for every button. So I can easily click on a user push 1 and boom he's timeouted for 1m (Spam).

In my opinion both options are good but I would prefer the **second** one because its easy to push one button instead of two or maybe three. I would love to welcome this feature very soon and I guess the realisation might be a little bit complicated with the first option.
I would love to see your opinion on this aswell.

Only one thing I have to add:
As @TranRed said, it is very channel specific and it would be nice if you could set those buttons and reasons specifically for one channel. But this is  something for the future to implement.

**Comment by NotEchoDE** [comment]
> An extension to timeout buttons specifying reasons might be cool. I decided to take this up and here's some basic concept of how it will look (let me know if it's good):
> https://cdn.zneix.eu/FU3ESlW.png
> ![THE TIGA IS IN DA CAGE](https://camo.githubusercontent.com/a3dfceaed0bab04faf29e5631eff7cf729b9d1c256e88d5e03100dc5bd72ec94/68747470733a2f2f63646e2e7a6e6569782e65752f46553345536c572e706e67)
> 
> For now, similar functionality can be achieved with custom moderation buttons (as seen on the screenshot above) - you can set those and those are accessible over either regular chat or usercard.

Is this build available everywehre? It would love to use it even tho its still in development.

**Comment by zneix** [comment]
@NotEchoDE nope, that was just some preview code I've written - it doesn't work properly, but I can finish it and make a pull request or maybe develop something else that will close this issue. I'm glad you like my suggestion and I will try to improve it!

**Comment by NotEchoDE** [comment]
Do you already know roughly know when this will be available?

**Comment by antonbowe** [comment]
Is this still in progress?
